war/src/main/js/tooltips.js

```diff
@@ -0,0 +1,91 @@
+import tippy from "tippy.js"
+
+const TOOLTIP_BASE = {
+  arrow: false,
+  theme: "tooltip",
+  animation: "tooltip",
+  appendTo: document.body
+}
+
+let tooltipInstances = []
+const globalPlugin = {
+  fn(instance) {
+    return {
+      onCreate() {
+        tooltipInstances = tooltipInstances.concat(instance)
+      },
+      onDestroy() {
+        tooltipInstances = tooltipInstances.filter(i => i !== instance)
+      }
+    }
+  }
+}
+
+tippy.setDefaultProps({
+  plugins: [globalPlugin]
+})
+
+registerTooltips()
+
+/**
+ * Registers tooltips for the page
+ * If called again, destroys existing tooltips and registers them again (useful for progressive rendering)
+ * @param {HTMLElement} container - Registers the tooltips for the given container
+ */
+function registerTooltips(container) {
+  if (!container) {
+    container = document
+  }
+
+  tooltipInstances.forEach(instance => {
+    if (instance.props.container === container) {
+      instance.destroy()
+    }
+  })
+
+  tippy(container.querySelectorAll("[tooltip]:not([tooltip=\"\"])"), Object.assign({
+    content: element => element.getAttribute("tooltip").replaceAll("<br>", "\n")
+      .replaceAll("<br/>", "\n")
+      .replaceAll("<br />", "\n")
+      .replaceAll("\\n", "\n"),
+    container: container,
+    onCreate(instance) {
+      instance.reference.setAttribute("title", instance.props.content)
+    },
+    onShow(instance) {
+      instance.reference.removeAttribute("title")
+    },
+    onHidden(instance) {
+      instance.reference.setAttribute("title", instance.props.content)
+    }
+  }, TOOLTIP_BASE))
+
+  tippy(container.querySelectorAll("[html-tooltip]"), Object.assign({
+    content: element => element.getAttribute("html-tooltip"),
+    allowHTML: true,
+    container: container,
+    interactive: true,
```

Comment:
Is there a reason the html version is interactive?

see https://github.com/jenkinsci/matrix-auth-plugin/pull/120#issuecomment-1115753934

It makes the matrix auth tooltips stick around somewhat annoyingly as you try click other ones near the current tooltip.

(if it needed for something like 'Warnings NG' we could probably make it configurable with data attributes?)

Reply:
This is no longer interactive 👍